Python/1-inner_class.py

```diff
@@ -0,0 +1,29 @@
+class Singletone:
+
+    class __Singletone:
+        def __init__(self, arg):
+
+            self.val = arg
+        def __str__(self):
+            return repr(self) + self.val
+
+    instance = None
+
+    def __init__(self, arg):
+        if not Singletone.instance:
+            Singletone.instance = Singletone.__Singletone(arg)
+        else:
+            Singletone.instance.val = arg
+
+    def __getattr__(self, name):
+        return getattr(self.instance, name)
+
+
```

Comment:
following `print statements` do not verify that `a, b, c` are singletones:
`a, b and c` have different addresses and they are different objects
only  `a.instance b.instance and c.instance` is a singleton